test/unified_malloc_framework/umf_pools/disjoint_pool.cpp

```diff
@@ -33,6 +33,16 @@ static auto makePool() {
     return std::move(pool);
 }
 
+static auto makePoolOOMProvider() {
```

Comment:
This function returns a pool, not a provider. Also, you could make this a template function and reuse it in the `proxyPoolOOMTest` test instantiation. It would look similar to `poolMakeUnique` where you accept `Args &&...args` as parameters and forward it to the pool `initialize` function.

Reply:
Done. I've added an extra template function.